source/mqtt_iot_sdk_metrics.c

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/encoding.h>
+#include <aws/common/string.h>
+#include <aws/common/system_info.h>
+#include <aws/mqtt/mqtt.h>
+#include <aws/mqtt/private/client_impl_shared.h>
+
+#include <stdio.h>
+
+// Maximum MQTT5 Content Type size https://docs.aws.amazon.com/general/latest/gr/iot-core.html#thing-limits
+const int AWS_IOT_MAX_CONTENT_SIZE = 256;
+
+int aws_mqtt_append_sdk_metrics_to_username(
+    struct aws_allocator *allocator,
+    const struct aws_byte_cursor *original_username,
+    const struct aws_mqtt_iot_sdk_metrics metrics,
```

Comment:
Why is this pass-by-value?  You could eliminate that caller complexity by making this a pointer and on null returning the base username or ""

Reply:
Debatable: as we have a username storage, I think it is redundant to do a memcopy and return the base username here.